compiler/rustc_attr_data_structures/src/lib.rs

```diff
@@ -1,3 +1,35 @@
+//! Data structures for representing parsed attributes in the Rust compiler.
```

Comment:
Once more, I won't merge this documentation here. Only a link to `rustc_attr_parsing` is all that's needed so we don't have to keep two sets of docs up to date.